multimedia/libcaption/Portfile

```diff
@@ -25,11 +27,16 @@ checksums           rmd160  bf5971f889ec0430bc8330140d6b08c25a790b2f \
                     sha256  6b0f2d1172ea38facc24d0ca5608bdf1ec1839745c88a0acba93a48106d4f4c1 \
                     size    140039
 
-# ld: duplicate symbol _caption_frame_timestamp in ../libcaption.a(srt.c.o) and CMakeFiles/scc2srt.dir/scc2srt.c.o
-compiler.blacklist-append *gcc-4.0 *gcc-4.2
+# ld: duplicate symbol _caption_frame_timestamp in
+# ../libcaption.a(srt.c.o) and CMakeFiles/scc2srt.dir/scc2srt.c.o
+compiler.blacklist-append   *gcc-4.0 *gcc-4.2
 
 configure.args-append       -DBUILD_EXAMPLES=OFF
 
+variant shared_lib description "Build ${name} as a shared library" {
+    configure.args-append   -DBUILD_SHARED_LIBS=ON
+}
```

Comment:
I am not convinced this should be a variant. Why not always do this - presumably you'll build both in this case (and even if not, we typically want shared libraries anyway).

Reply:
The upstream developer's code only builds a static library by default. I was just exposing the option to users in case they want to choose, but I suppose that as a package management system, we're only really interested in a shared library.

Reply:
Okay, I've made building as a shared lib mandatory.